nifi-frontend/src/main/frontend/apps/nifi/src/app/pages/flow-designer/ui/canvas/items/processor/edit-processor/edit-processor.component.html

```diff
@@ -17,17 +17,60 @@
 
 <h2 mat-dialog-title>
     <div class="flex justify-between items-baseline">
-        <div>
-            {{ readonly ? 'Processor Details' : 'Edit Processor' }}
+        <div class="flex items-baseline">
+            <div class="mr-2">
+                @if (isInvalid(request.entity)) {
+                    <i
+                        class="mr-2 fa fa-warning caution-color"
+                        nifiTooltip
+                        [tooltipComponentType]="ValidationErrorsTip"
+                        [tooltipInputData]="getValidationErrorsTipData(request.entity)"></i>
+                }{{ readonly ? 'Processor Details' : 'Edit Processor' }}
+            </div>
+            |
+            <div class="ml-2 text-base">
+                {{ formatType(request.entity) }}
+            </div>
         </div>
-        <div class="text-base">
-            {{ formatType(request.entity) }}
+        <div>
+            @if (isStoppable(request.entity)) {
+                <button type="button" mat-stroked-button [matMenuTriggerFor]="operateMenu">
+                    <div class="flex items-center">
+                        <i class="mr-2 success-color-default fa fa-play"></i>Running<i
+                            class="ml-2 -mt-1 fa fa-sort-desc"></i>
+                    </div>
+                </button>
+            } @else if (isRunnable(request.entity)) {
+                <button type="button" mat-stroked-button [matMenuTriggerFor]="operateMenu">
+                    <div class="flex items-center">
+                        <i class="mr-2 error-color-variant fa fa-stop"></i>Stopped<i
+                            class="ml-2 -mt-1 fa fa-sort-desc"></i>
+                    </div>
+                </button>
+            } @else {
+                <button type="button" mat-stroked-button [disabled]="true">
+                    <div class="flex items-center">
+                        @if (!isInvalid(request.entity)) {
+                            <i class="mr-2 fa fa-circle-o-notch fa-spin primary-color"></i>
+                        }
+                        {{ formatRunStatus() }}
+                    </div>
+                </button>
+            }
+            <mat-menu #operateMenu="matMenu" xPosition="before">
+                @if (isStoppable(request.entity)) {
+                    <button mat-menu-item (click)="stop(request.entity)">Stop</button>
+                } @else if (isRunnable(request.entity)) {
+                    <button mat-menu-item [disabled]="editProcessorForm.dirty" (click)="start(request.entity)">
+                        Start
+                    </button>
+                }
```

Comment:
I think we should provide the start/stop icons in the menu just as we do in the canvas context menu.

Reply:
<img width="144" alt="Screenshot 2024-11-25 at 11 57 23 AM" src="https://github.com/user-attachments/assets/a99a406d-858c-4d44-b269-6c90a6081954">